{"uid":"a4863d10c245c5dc","name":"test_record_d365_authentication","fullName":"tests.d365.test_d365_auth#test_record_d365_authentication","historyId":"92b47c5c72be6742c31b5c1edf35d301","time":{"start":1761791545539,"stop":1761791666195,"duration":120656},"description":"\n    Record D365 authentication session.\n    \n    IMPORTANT: Run this test manually with headed mode:\n        pytest tests/d365/test_d365_auth.py -k record --headed -s\n    \n    Steps:\n        1. Navigate to D365\n        2. Login manually\n        3. Wait for dashboard to load\n        4. Session is saved automatically\n    ","descriptionHtml":"<pre><code>Record D365 authentication session.\n\nIMPORTANT: Run this test manually with headed mode:\n    pytest tests/d365/test_d365_auth.py -k record --headed -s\n\nSteps:\n    1. Navigate to D365\n    2. Login manually\n    3. Wait for dashboard to load\n    4. Session is saved automatically\n</code></pre>\n","status":"failed","statusMessage":"Failed: Authentication timeout. Please login within 2 minutes. Error: Not on D365 dashboard yet. Current URL: https://login.microsoftonline.com/ba4f0c17-6451-450e-8adb-a7df399faaa9/oauth2/authorize?client_id=00000015-0000-0000-c000-000000000000&response_type=code%20id_token&scope=openid%20profile&state=OpenIdConnect.AuthenticationProperties%3DAQAAANCMnd8BFdERjHoAwE_Cl-sBAAAAD_dUx6Kg1kW6IzAC2a1VHQAAAAACAAAAAAAQZgAAAAEAACAAAAA0KGLZ2g_EPdDfQkf_kbsJ-NdmaZQsvpsstl108fhhzgAAAAAOgAAAAAIAACAAAAAgIVG8bemojOTo-dq4FRk6Vnw2LV7rIB2yVkoOEfhtf3AAAAASYWEraK-p6EZ1n3EOB_gNayS6LUWORTrz7owSyuGUSVkvv9kGvZhwmjpU_NtgcvdrK8IYor334QpeOxO6duVEdYAAlkfzTiVRk-nzurMQzTTIS0hgIDGrDTYqR3Dk-OaWOBljZQMKBRh6VT2LMwjAQAAAAOlrwmRo5YirpD0A-sPpxdasFvqyy6jBtVUGWWjjl515K1lb4GjfBeYSJ8_XEwhAZe4mM-jn90xbyKX2vgiE3W0%26RedirectTo%3Dhttps%253A%252F%252Ffourhands-test.sandbox.operations.dynamics.com%252F&response_mode=form_post&nonce=638973883452969555.Njc4YzVkY2ItYTYwMC00NGEzLWI4ZDktN2IwYWJiYzQ0Y2MxOTcwNDk3MDItZGRkNS00MmM2LWE2MGMtZDNjNzA3MTZjNGYw&redirect_uri=https%3A%2F%2Ffourhands-test.sandbox.operations.dynamics.com%2F&max_age=86400&x-client-SKU=ID_NET472&x-client-ver=8.3.0.0","statusTrace":"tests/d365/test_d365_auth.py:55: in test_record_d365_authentication\n    raise Exception(f\"Not on D365 dashboard yet. Current URL: {current_url}\")\nE   Exception: Not on D365 dashboard yet. Current URL: https://login.microsoftonline.com/ba4f0c17-6451-450e-8adb-a7df399faaa9/oauth2/authorize?client_id=00000015-0000-0000-c000-000000000000&response_type=code%20id_token&scope=openid%20profile&state=OpenIdConnect.AuthenticationProperties%3DAQAAANCMnd8BFdERjHoAwE_Cl-sBAAAAD_dUx6Kg1kW6IzAC2a1VHQAAAAACAAAAAAAQZgAAAAEAACAAAAA0KGLZ2g_EPdDfQkf_kbsJ-NdmaZQsvpsstl108fhhzgAAAAAOgAAAAAIAACAAAAAgIVG8bemojOTo-dq4FRk6Vnw2LV7rIB2yVkoOEfhtf3AAAAASYWEraK-p6EZ1n3EOB_gNayS6LUWORTrz7owSyuGUSVkvv9kGvZhwmjpU_NtgcvdrK8IYor334QpeOxO6duVEdYAAlkfzTiVRk-nzurMQzTTIS0hgIDGrDTYqR3Dk-OaWOBljZQMKBRh6VT2LMwjAQAAAAOlrwmRo5YirpD0A-sPpxdasFvqyy6jBtVUGWWjjl515K1lb4GjfBeYSJ8_XEwhAZe4mM-jn90xbyKX2vgiE3W0%26RedirectTo%3Dhttps%253A%252F%252Ffourhands-test.sandbox.operations.dynamics.com%252F&response_mode=form_post&nonce=638973883452969555.Njc4YzVkY2ItYTYwMC00NGEzLWI4ZDktN2IwYWJiYzQ0Y2MxOTcwNDk3MDItZGRkNS00MmM2LWE2MGMtZDNjNzA3MTZjNGYw&redirect_uri=https%3A%2F%2Ffourhands-test.sandbox.operations.dynamics.com%2F&max_age=86400&x-client-SKU=ID_NET472&x-client-ver=8.3.0.0\n\nDuring handling of the above exception, another exception occurred:\ntests/d365/test_d365_auth.py:58: in test_record_d365_authentication\n    pytest.fail(f\"Authentication timeout. Please login within 2 minutes. Error: {e}\")\nE   Failed: Authentication timeout. Please login within 2 minutes. Error: Not on D365 dashboard yet. Current URL: https://login.microsoftonline.com/ba4f0c17-6451-450e-8adb-a7df399faaa9/oauth2/authorize?client_id=00000015-0000-0000-c000-000000000000&response_type=code%20id_token&scope=openid%20profile&state=OpenIdConnect.AuthenticationProperties%3DAQAAANCMnd8BFdERjHoAwE_Cl-sBAAAAD_dUx6Kg1kW6IzAC2a1VHQAAAAACAAAAAAAQZgAAAAEAACAAAAA0KGLZ2g_EPdDfQkf_kbsJ-NdmaZQsvpsstl108fhhzgAAAAAOgAAAAAIAACAAAAAgIVG8bemojOTo-dq4FRk6Vnw2LV7rIB2yVkoOEfhtf3AAAAASYWEraK-p6EZ1n3EOB_gNayS6LUWORTrz7owSyuGUSVkvv9kGvZhwmjpU_NtgcvdrK8IYor334QpeOxO6duVEdYAAlkfzTiVRk-nzurMQzTTIS0hgIDGrDTYqR3Dk-OaWOBljZQMKBRh6VT2LMwjAQAAAAOlrwmRo5YirpD0A-sPpxdasFvqyy6jBtVUGWWjjl515K1lb4GjfBeYSJ8_XEwhAZe4mM-jn90xbyKX2vgiE3W0%26RedirectTo%3Dhttps%253A%252F%252Ffourhands-test.sandbox.operations.dynamics.com%252F&response_mode=form_post&nonce=638973883452969555.Njc4YzVkY2ItYTYwMC00NGEzLWI4ZDktN2IwYWJiYzQ0Y2MxOTcwNDk3MDItZGRkNS00MmM2LWE2MGMtZDNjNzA3MTZjNGYw&redirect_uri=https%3A%2F%2Ffourhands-test.sandbox.operations.dynamics.com%2F&max_age=86400&x-client-SKU=ID_NET472&x-client-ver=8.3.0.0","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"_verify_url","time":{"start":1761791542172,"stop":1761791542172,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"pytestconfig","time":{"start":1761791542172,"stop":1761791542173,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"base_url","time":{"start":1761791542172,"stop":1761791542172,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"playwright","time":{"start":1761791542173,"stop":1761791542479,"duration":306},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"delete_output_dir","time":{"start":1761791542173,"stop":1761791542173,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"playwright_browser","time":{"start":1761791542479,"stop":1761791545476,"duration":2997},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"browser_context_args","time":{"start":1761791545477,"stop":1761791545477,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"second_fixture","time":{"start":1761791545477,"stop":1761791545477,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"caplog","time":{"start":1761791545477,"stop":1761791545477,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"configure_browserstack","time":{"start":1761791545478,"stop":1761791545478,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"context","time":{"start":1761791545478,"stop":1761791545493,"duration":15},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"page","time":{"start":1761791545493,"stop":1761791545538,"duration":45},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"description":"\n    Record D365 authentication session.\n    \n    IMPORTANT: Run this test manually with headed mode:\n        pytest tests/d365/test_d365_auth.py -k record --headed -s\n    \n    Steps:\n        1. Navigate to D365\n        2. Login manually\n        3. Wait for dashboard to load\n        4. Session is saved automatically\n    ","status":"failed","statusMessage":"Failed: Authentication timeout. Please login within 2 minutes. Error: Not on D365 dashboard yet. Current URL: https://login.microsoftonline.com/ba4f0c17-6451-450e-8adb-a7df399faaa9/oauth2/authorize?client_id=00000015-0000-0000-c000-000000000000&response_type=code%20id_token&scope=openid%20profile&state=OpenIdConnect.AuthenticationProperties%3DAQAAANCMnd8BFdERjHoAwE_Cl-sBAAAAD_dUx6Kg1kW6IzAC2a1VHQAAAAACAAAAAAAQZgAAAAEAACAAAAA0KGLZ2g_EPdDfQkf_kbsJ-NdmaZQsvpsstl108fhhzgAAAAAOgAAAAAIAACAAAAAgIVG8bemojOTo-dq4FRk6Vnw2LV7rIB2yVkoOEfhtf3AAAAASYWEraK-p6EZ1n3EOB_gNayS6LUWORTrz7owSyuGUSVkvv9kGvZhwmjpU_NtgcvdrK8IYor334QpeOxO6duVEdYAAlkfzTiVRk-nzurMQzTTIS0hgIDGrDTYqR3Dk-OaWOBljZQMKBRh6VT2LMwjAQAAAAOlrwmRo5YirpD0A-sPpxdasFvqyy6jBtVUGWWjjl515K1lb4GjfBeYSJ8_XEwhAZe4mM-jn90xbyKX2vgiE3W0%26RedirectTo%3Dhttps%253A%252F%252Ffourhands-test.sandbox.operations.dynamics.com%252F&response_mode=form_post&nonce=638973883452969555.Njc4YzVkY2ItYTYwMC00NGEzLWI4ZDktN2IwYWJiYzQ0Y2MxOTcwNDk3MDItZGRkNS00MmM2LWE2MGMtZDNjNzA3MTZjNGYw&redirect_uri=https%3A%2F%2Ffourhands-test.sandbox.operations.dynamics.com%2F&max_age=86400&x-client-SKU=ID_NET472&x-client-ver=8.3.0.0","statusTrace":"tests/d365/test_d365_auth.py:55: in test_record_d365_authentication\n    raise Exception(f\"Not on D365 dashboard yet. Current URL: {current_url}\")\nE   Exception: Not on D365 dashboard yet. Current URL: https://login.microsoftonline.com/ba4f0c17-6451-450e-8adb-a7df399faaa9/oauth2/authorize?client_id=00000015-0000-0000-c000-000000000000&response_type=code%20id_token&scope=openid%20profile&state=OpenIdConnect.AuthenticationProperties%3DAQAAANCMnd8BFdERjHoAwE_Cl-sBAAAAD_dUx6Kg1kW6IzAC2a1VHQAAAAACAAAAAAAQZgAAAAEAACAAAAA0KGLZ2g_EPdDfQkf_kbsJ-NdmaZQsvpsstl108fhhzgAAAAAOgAAAAAIAACAAAAAgIVG8bemojOTo-dq4FRk6Vnw2LV7rIB2yVkoOEfhtf3AAAAASYWEraK-p6EZ1n3EOB_gNayS6LUWORTrz7owSyuGUSVkvv9kGvZhwmjpU_NtgcvdrK8IYor334QpeOxO6duVEdYAAlkfzTiVRk-nzurMQzTTIS0hgIDGrDTYqR3Dk-OaWOBljZQMKBRh6VT2LMwjAQAAAAOlrwmRo5YirpD0A-sPpxdasFvqyy6jBtVUGWWjjl515K1lb4GjfBeYSJ8_XEwhAZe4mM-jn90xbyKX2vgiE3W0%26RedirectTo%3Dhttps%253A%252F%252Ffourhands-test.sandbox.operations.dynamics.com%252F&response_mode=form_post&nonce=638973883452969555.Njc4YzVkY2ItYTYwMC00NGEzLWI4ZDktN2IwYWJiYzQ0Y2MxOTcwNDk3MDItZGRkNS00MmM2LWE2MGMtZDNjNzA3MTZjNGYw&redirect_uri=https%3A%2F%2Ffourhands-test.sandbox.operations.dynamics.com%2F&max_age=86400&x-client-SKU=ID_NET472&x-client-ver=8.3.0.0\n\nDuring handling of the above exception, another exception occurred:\ntests/d365/test_d365_auth.py:58: in test_record_d365_authentication\n    pytest.fail(f\"Authentication timeout. Please login within 2 minutes. Error: {e}\")\nE   Failed: Authentication timeout. Please login within 2 minutes. Error: Not on D365 dashboard yet. Current URL: https://login.microsoftonline.com/ba4f0c17-6451-450e-8adb-a7df399faaa9/oauth2/authorize?client_id=00000015-0000-0000-c000-000000000000&response_type=code%20id_token&scope=openid%20profile&state=OpenIdConnect.AuthenticationProperties%3DAQAAANCMnd8BFdERjHoAwE_Cl-sBAAAAD_dUx6Kg1kW6IzAC2a1VHQAAAAACAAAAAAAQZgAAAAEAACAAAAA0KGLZ2g_EPdDfQkf_kbsJ-NdmaZQsvpsstl108fhhzgAAAAAOgAAAAAIAACAAAAAgIVG8bemojOTo-dq4FRk6Vnw2LV7rIB2yVkoOEfhtf3AAAAASYWEraK-p6EZ1n3EOB_gNayS6LUWORTrz7owSyuGUSVkvv9kGvZhwmjpU_NtgcvdrK8IYor334QpeOxO6duVEdYAAlkfzTiVRk-nzurMQzTTIS0hgIDGrDTYqR3Dk-OaWOBljZQMKBRh6VT2LMwjAQAAAAOlrwmRo5YirpD0A-sPpxdasFvqyy6jBtVUGWWjjl515K1lb4GjfBeYSJ8_XEwhAZe4mM-jn90xbyKX2vgiE3W0%26RedirectTo%3Dhttps%253A%252F%252Ffourhands-test.sandbox.operations.dynamics.com%252F&response_mode=form_post&nonce=638973883452969555.Njc4YzVkY2ItYTYwMC00NGEzLWI4ZDktN2IwYWJiYzQ0Y2MxOTcwNDk3MDItZGRkNS00MmM2LWE2MGMtZDNjNzA3MTZjNGYw&redirect_uri=https%3A%2F%2Ffourhands-test.sandbox.operations.dynamics.com%2F&max_age=86400&x-client-SKU=ID_NET472&x-client-ver=8.3.0.0","steps":[],"attachments":[{"uid":"7b95f296f5565b77","name":"Screenshot - test_record_d365_authentication","source":"7b95f296f5565b77.png","type":"image/png","size":166987},{"uid":"d23c9e5d62a24b5e","name":"Page HTML","source":"d23c9e5d62a24b5e.html","type":"text/html","size":85381},{"uid":"3f67ff67730ce850","name":"Page URL","source":"3f67ff67730ce850.txt","type":"text/plain","size":1038},{"uid":"7623ffe2f8a52efa","name":"Test Video","source":"7623ffe2f8a52efa.webm","type":"video/webm","size":6336183},{"uid":"5aa9be82fdbb9292","name":"stdout","source":"5aa9be82fdbb9292.txt","type":"text/plain","size":532}],"parameters":[],"stepsCount":0,"shouldDisplayMessage":true,"attachmentsCount":5,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"page::0","time":{"start":1761791667522,"stop":1761791667524,"duration":2},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"context::0","time":{"start":1761791667524,"stop":1761791667526,"duration":2},"status":"broken","statusMessage":"playwright._impl._errors.TargetClosedError: Tracing.stop: Target page, context or browser has been closed\n","statusTrace":"  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 231, in __call__\n    return self._fixture_function(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/_pytest/fixtures.py\", line 939, in _teardown_yield_fixture\n    next(it)\n  File \"/home/runner/work/QA-playwright-automation/QA-playwright-automation/conftest.py\", line 118, in context\n    context.tracing.stop(path=trace_path)\n  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/playwright/sync_api/_generated.py\", line 15237, in stop\n    return mapping.from_maybe_impl(self._sync(self._impl_obj.stop(path=path)))\n                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/playwright/_impl/_sync_base.py\", line 115, in _sync\n    return task.result()\n           ^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/playwright/_impl/_tracing.py\", line 68, in stop\n    await self._do_stop_chunk(path)\n  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/playwright/_impl/_tracing.py\", line 87, in _do_stop_chunk\n    result = await self._channel.send_return_as_dict(\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/playwright/_impl/_connection.py\", line 83, in send_return_as_dict\n    return await self._connection.wrap_api_call(\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/playwright/_impl/_connection.py\", line 558, in wrap_api_call\n    raise rewrite_error(error, f\"{parsed_st['apiName']}: {error}\") from None\n","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":true,"attachmentsCount":0,"hasContent":true,"attachmentStep":false},{"name":"second_fixture::0","time":{"start":1761791667527,"stop":1761791667527,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"caplog::0","time":{"start":1761791667527,"stop":1761791667527,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"playwright_browser::0","time":{"start":1761791858576,"stop":1761791858615,"duration":39},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"playwright::0","time":{"start":1761791858616,"stop":1761791858623,"duration":7},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"tag","value":"record"},{"name":"tag","value":"record"},{"name":"parentSuite","value":"tests.d365"},{"name":"suite","value":"test_d365_auth"},{"name":"host","value":"runnervmwhb2z"},{"name":"thread","value":"2634-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.d365.test_d365_auth"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Product defects","matchedStatuses":[],"flaky":false}],"tags":["record"]},"source":"a4863d10c245c5dc.json","parameterValues":[]}